packages/pages/src/common/src/template/internal/getTemplateFilepaths.ts

```diff
@@ -0,0 +1,37 @@
+import glob from "glob";
+import path from "path";
+import { Path } from "../../project/path.js";
+
+/**
+ * Get all the template files in the provided template folder path(s).
+ * If there are two files that share the same name between the two
+ * provided template folder paths, only the file in domain template
+ * folder path is collected.
+ *
+ * @param templatesRoot  The folder path where all template files live
+ * @param templatesDomain The folder path where the template files of a specific domain live
+ * @returns a list of template filepaths
+ */
+export const getTemplateFilepaths = (
```

Comment:
I don't think that getTemplateFilepaths should know about this scoping logic -- that should be relatively contained. What if it just took a list of paths, iterated through them in-order, and only kept the last seen templates?

Reply:
Hmm, not sure how I feel about having the "order matter" logic when processing the list of paths.. could be a little confusing but that should be fine with documentation.
Just to get a better understanding, what is the reason you think it shouldn't know about about domain / root? Since it's a concept we are introducing, I think this help makes it more clear that domain files take precedent over root files, and any unique root files are share across domains.

Reply:
My main goal is to limit how much other components needs to handle specific scoping logic. Ideally, it'd only be contained within the ProjectStructure.

I don't see why `getTemplateFilepaths` needs to have custom logic on whether a domain was set or not. Instead it can just accept an arbitrary number of paths with rules that match the business rules of how to determine which template is which.

Reply:
fair point, I will update!